### ISSUE TITLE
Error involving inlined code

### ISSUE THREAD

**Issue by julienrf** [lampepfl/dotty#2404]
The [following code](https://github.com/scala/collection-strawman/blob/7af8b174b251579a3208f0ae61bcf237ed0085be/src/main/scala/strawman/collection/Map.scala#L81-L84) fails to compile with the following error:

~~~
[error] 84 |  @`inline` final def ++ [V2 >: V](xs: collection.Iterable[(K, V2)]): CC[K, V2] = concat(xs)
[error]    |                                                                                  ^^^^^^
[error]    |value of type <nonsensical><notype></nonsensical> does not take type parameters
[error]    | This location is in code that was inlined at /home/travis/build/scala/collection-strawman/src/main/scala/strawman/collection/immutable/HashMap.scala:106
~~~

I’ll try to minimize the test case.

**Comment by julienrf** [comment]
It seems that the error is not due to inlining but to overloading, and it seems to be fixed by #2408.

**Comment by odersky** [comment]
I think it was a combination of inlining and overloading. Overloading resolution depends on signatures, and signatures were broken by inlining.
